### PR TITLE
feat(web3hub): add confirmation bottom modal before opening app [LIVE-13188]

### PR DESCRIPTION
**Pull request by Justkant**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - Web3Hub (behind FF)

### 📝 Description

Add confirmation bottom modal before opening app
Updated the base url used to get manifest in web3hub
Refactor a bit our flashlist usage
Updated tests

| Before        | After         |
| ------------- | ------------- |
| nothing | Screenshot 2024-08-01 at 20 10 38 |

### ❓ Context

- **JIRA or GitHub link**: [LIVE-13188]

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-13188]: https://ledgerhq.atlassian.net/browse/LIVE-13188?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-mobile/src/newArch/features/Web3Hub/utils/api/manifests.ts [hunk 1]
```diff
@@ -36,8 +37,19 @@ export const fetchManifests: (
 ) => QueryFunction<LiveAppManifest[], string[], number> =
   (category, search) =>
   async ({ pageParam }) => {
+    const url = new URL(`${URL_ORIGIN}/api/v2/apps`);
```
Comment (KVNLS) [inline apps/ledger-live-mobile/src/newArch/features/Web3Hub/utils/api/manifests.ts:40]: We introduced react-query on both LLM & LLD, so feel free to use it if you need it.

**Review by KVNLS — APPROVED** [review]

File: apps/ledger-live-mobile/src/newArch/features/Web3Hub/utils/api/manifests.ts (hunk 1 above)
Comment (Justkant) [reply apps/ledger-live-mobile/src/newArch/features/Web3Hub/utils/api/manifests.ts:40]: Nice thx for the heads up

**Review by mcayuelas-ledger — APPROVED** [review]

File: apps/ledger-live-mobile/src/newArch/features/Web3Hub/components/Disclaimer/index.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      ) : null}
+
+      <Box mt={6} height="1px" width="100%" backgroundColor={"translucentGrey"} />
+
+      <Flex mt={6} flexDirection={"row"} alignItems={"center"}>
+        <Box mr={2}>
+          <Icons.Eye color={"smoke"} />
+        </Box>
+        <Text fontSize={14} color="smoke">
+          {t("web3hub.disclaimer.clearSigningEnabled")}
```
Comment (RamyEB) [inline apps/ledger-live-mobile/src/newArch/features/Web3Hub/components/Disclaimer/index.tsx:51]: We should display that in certains conditions

**Review by RamyEB — CHANGES_REQUESTED** [review]

File: apps/ledger-live-mobile/src/newArch/features/Web3Hub/components/Disclaimer/index.tsx (hunk 2 above)
Comment (Justkant) [reply apps/ledger-live-mobile/src/newArch/features/Web3Hub/components/Disclaimer/index.tsx:51]: We already have ticket when the category is available https://ledgerhq.atlassian.net/browse/LIVE-13569

File: apps/ledger-live-mobile/src/newArch/features/Web3Hub/utils/api/manifests.ts (hunk 1 above)
Comment (Justkant) [reply apps/ledger-live-mobile/src/newArch/features/Web3Hub/utils/api/manifests.ts:40]: Actually I'm already using it here the `QueryFunction ` for example comes from it, so you would it help for this ?

**Pull request merged**